src/attribute/Initializable.sol

```diff
@@ -1,33 +1,38 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.13;
 
-import "./interfaces/IInitializable.sol";
+import { IInitializable } from "./interfaces/IInitializable.sol";
 
 /// @title Initializable
 /// @notice Library to manage the initialization lifecycle of upgradeable contracts
 /// @dev `Initializable.sol` allows the creation of pseudo-constructors for upgradeable contracts. One
 ///      `initializer` should be declared per top-level contract. Child contracts can use the `onlyInitializer`
 ///      modifier to tag their internal initialization functions to ensure that they can only be called
 ///      from a top-level `initializer` or a constructor.
+///      Name and Version are used by Proxy to validate contract upgrades.
 abstract contract Initializable is IInitializable {
-    /// @dev The initialized flag
-    uint256 private _version;
+    /// @dev Identifies the contract
+    string public name;
+
+    /// @dev Nonzero indicates contract is initialized
+    uint256 public version;
 
     /// @dev The initializing flag
     bool private _initializing;
 
     /// @dev Can only be called once per version, `version` is 1-indexed
-    modifier initializer(uint256 version) {
-        if (version == 0) revert InitializableZeroVersionError();
-        if (_version >= version) revert InitializableAlreadyInitializedError(version);
+    /// @param name_ Identifies the contract, must not change
+    modifier initializer(string memory name_, uint256 version_) {
+        if (version_ == 0) revert InitializableZeroVersionError();
```

Comment:
We can probably forgo any version check in here.

Reply:
Done in 6f03445cbecec00d2abd9e5f6c4505ce07734cd3.